File: tests/FSharpAux.Mocha.Tests/fable/src/FSharpAux.Core/Set.js
import { intersect as intersect_1, union as union_1, difference } from "../../fable_modules/fable-library.4.0.6/Set.js";

export function symmetricDifference(set1, set2) {
    return difference(union_1(set1, set2), intersect_1(set1, set2));
}

File: src/FSharpAux.Core/Set.fs
namespace FSharpAux


[<AutoOpen>]
module Set =    

    ///Computes the symmteric difference of the two sets, which is the set of elements which are in either of the sets and not in their intersection
    let symmetricDifference (set1:Set<'T>) (set2:Set<'T>) =

        let union = Set.union set1 set2
        let intersect = Set.intersect set1 set2

        Set.difference union intersect